src/core/utils/color/ColorValue.ts

```diff
@@ -649,25 +676,36 @@ export class ColorValue {
     return this._tbgr === other._tbgr && this._hue === other._hue;
   }
 
+  public static getFormattedColorNumber(value: number, precision = 2) {
```

Comment:
This still doesn't fix the problem where `ColorValue.equals()` doesn't work correctly. 

In line 164 of ColorPalette.tsx, it is setting the `ColorSwatch` to active using `isActive={ color.toHslString(true) === activeColor.toHslString(true)}`, which works ok for now, but ideally we want to use `isActive={color.equals(activeColor)}`. 

However, when using is `equals` the active color is not set correctly when the input is changed:
![isActive](https://user-images.githubusercontent.com/88331924/141150970-0c0b895d-6dda-4c75-a826-56515dfa9df3.gif)

Can you look to see if there's a way to use equals instead of toHslString?

Reply:
The latest updates use .equals instead of comparing strings.

Reply:
Yeah but it still has the same bug from above where the color swatch is not set to active even when the input should match. Maybe we should just leave comparing strings?

Reply:
try this again after latest push.